setPL.sh

```diff
@@ -205,20 +211,26 @@ verifyTurbostatInstallation
 echo "**** Current PL values from 'turbostat'"
 printTurboStat_PL1_PL2
 
-# set new PL1/PL2 values
-echo "**** Setting PL1=$PL1 and PL2=$PL2 in /sys/class/powercap/intel-rapl/intel-rapl:0/constraint_*_power_limit_uw"
-echo "$PL1" > /sys/class/powercap/intel-rapl/intel-rapl:0/constraint_0_power_limit_uw
-echo "$PL2" > /sys/class/powercap/intel-rapl/intel-rapl:0/constraint_1_power_limit_uw
+if [ $readOnly -ne $TRUE ]; then
+    # set new PL1/PL2 values
+    echo "**** Setting PL1=$PL1 and PL2=$PL2 in /sys/class/powercap/intel-rapl/intel-rapl:0/constraint_*_power_limit_uw"
+    echo "$PL1" > /sys/class/powercap/intel-rapl/intel-rapl:0/constraint_0_power_limit_uw
+    echo "$PL2" > /sys/class/powercap/intel-rapl/intel-rapl:0/constraint_1_power_limit_uw
+fi
 
 # enable both PL1 and PL2 (bit 15 and 47) if not already enabled
 readMsr $INTEL_MSR_PKG_POWER_LIMIT; msr=$retVal
 is_PL1_PL2_Enabled=$(((msr & INTEL_PL1_PL2_ENABLE_BITS) == INTEL_PL1_PL2_ENABLE_BITS))
 if [ $is_PL1_PL2_Enabled -ne $TRUE ]; then
-    echo "**** Enabling PL1 and PL2 in MSR_PKG_POWER_LIMIT"
-    msr=$((msr | INTEL_PL1_PL2_ENABLE_BITS))
-    writeMsr $INTEL_MSR_PKG_POWER_LIMIT $msr
+    if [ $readOnly -ne $TRUE ]; then
+        echo "**** Enabling PL1 and PL2 in MSR_PKG_POWER_LIMIT"
+        msr=$((msr | INTEL_PL1_PL2_ENABLE_BITS))
+        writeMsr $INTEL_MSR_PKG_POWER_LIMIT $msr
+    else
+        echo "**** PL1 and PL2 are DISabled in MSR_PKG_POWER_LIMIT"
```

Comment:
turbostat's strange capitalization of "DISabled" and "ENabled" doesn't need to be matched in setPL.